harness/determined/exec/launch.py

```diff
@@ -2,13 +2,30 @@
 import json
 import logging
 import os
+import signal
 import subprocess
 import sys
+import types
 from typing import Dict
 
 import determined as det
 import determined.common
-from determined.common import constants, storage
+from determined.common import api, constants, storage
+from determined.exec import prep_container
+
+
+# Signal handler to intercept SIGTERM notification of pending preemption
+def trigger_preemption(signum: int, frame: types.FrameType) -> None:
+    info = det.get_cluster_info()
+    if info and info.container_rank == 0:
+        # Chief container, requests preemption, others ignore
+        logging.debug(
+            f"[rank={info.container_rank}] SIGTERM: Preemption imminent. Triggering Checkpoint."
```

Comment:
"Triggering Checkpoint" is a bit misleading since we may finish some training first (and when we do it we will log anyway).
```suggestion
            f"[rank={info.container_rank}] SIGTERM: Preemption imminent."
```